spec/requests/email_domain/delete_spec.rb

```diff
@@ -1,5 +1,11 @@
 describe "DELETE /authorised_email_domains/:id", type: :request do
   let!(:email_domain) { create(:authorised_email_domain, name: 'some.domain.org.uk') }
+  let(:gateway_instance) { instance_double(Gateways::S3) }
+
+  before do
+    allow(Gateways::S3).to receive(:new).and_return(gateway_instance)
+    allow(gateway_instance).to receive(:write)
```

Comment:
Having this `allow(gateway_instance).to receive(:write)` and also `expect(gateway_instance).to have_received(:write)` on line 24 seems odd. Like we are telling rspec to do something and then testing it did it.

Reply:
A better approach would be something like:

L3: `let(:gateway) { instance_spy(Gateways::S3, write: nil) }`
Delete L7
L24: `expect(gateway).to have_received(:write)`